streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableKTableLeftJoinTest.java

```diff
@@ -94,80 +89,73 @@ public void testJoin() {
 
         final KTableValueGetterSupplier<Integer, String> getterSupplier = ((KTableImpl<Integer, String, String>) joined).valueGetterSupplier();
 
-        driver.setUp(builder, stateDir);
-        driver.setTime(0L);
+        try (final TopologyTestDriverWrapper driver = new TopologyTestDriverWrapper(builder.build(), props)) {
 
-        final MockProcessor<Integer, String> processor = supplier.theCapturedProcessor();
+            final MockProcessor<Integer, String> processor = supplier.theCapturedProcessor();
 
-        final KTableValueGetter<Integer, String> getter = getterSupplier.get();
-        getter.init(driver.context());
+            final KTableValueGetter<Integer, String> getter = getterSupplier.get();
+            getter.init(driver.getProcessorContext(topic1));
 
-        // push two items to the primary stream. the other table is empty
+            // push two items to the primary stream. the other table is empty
 
-        for (int i = 0; i < 2; i++) {
-            driver.process(topic1, expectedKeys[i], "X" + expectedKeys[i]);
-        }
-        // pass tuple with null key, it will be discarded in join process
-        driver.process(topic1, null, "SomeVal");
-        driver.flushState();
+            for (int i = 0; i < 2; i++) {
+                driver.pipeInput(recordFactory.create(topic1, expectedKeys[i], "X" + expectedKeys[i]));
+            }
+            // pass tuple with null key, it will be discarded in join process
+            driver.pipeInput(recordFactory.create(topic1, null, "SomeVal"));
 
-        processor.checkAndClearProcessResult("0:X0+null", "1:X1+null");
-        checkJoinedValues(getter, kv(0, "X0+null"), kv(1, "X1+null"), kv(2, null), kv(3, null));
+            processor.checkAndClearProcessResult("0:X0+null", "1:X1+null");
```

Comment:
Wondering, if we should get rid of `MockProcessor` and update the test setup to pipe the result into a topic instead. Than we would use `OutputVerifier` instead of dealing with all those concatenated Strings.

Reply:
That sounds way cleaner. If it's ok I would like to implement that in a subsequent PR, to avoid cramming even more changes into this one.